core/storage/job/gae_models.py

```diff
@@ -143,3 +143,55 @@ class ContinuousComputationModel(base_models.BaseModel):
     # The time at which a halt signal was last sent to this batch job, in
     # milliseconds since the epoch.
     last_stopped_msec = ndb.FloatProperty(indexed=True)
+
+
+# Allowed transitions: processing --> completed --> archived
+# or processing --> failed.
+QUERY_STATUS_PROCESSING = "processing"
+QUERY_STATUS_COMPLETED = "completed"
+QUERY_STATUS_ARCHIVED = "archived"
+QUERY_STATUS_FAILED = "failed"
+
+
+# Keys for query_parameters dictionary.
+# Values corresponding to key is value specified by query submitter.
+# A query can have any combination of following parameters.
+# Valid parameters names.
```

Comment:
I suggest replacing all the above with: "The possible options for a user query. (Note that a user query can include any combination of these options.)"

(You can talk about values at a place in the code where there actually are values.)